wordpress-mobile/WordPress-Android
# Dashboard Personalization: Activity Card implement hide card

Parent #18948 

This PR implements the "hide this" actions for the Recent Activity dashboard card
- Persists the "hide" state of Recent Activity dashboard card
- Refreshes the view state on "hide this" action
- Uses the hide preference to determine if the card should be shown

Merge Instructions
- Ensure #19005 has been merged
- Remove Do Not Merge Label
- Merge as normal

**To test:**
- Install the app
- Login and chose a site that has access to Activity Log
- Navigate to My Site tab
- Scroll down until you reach the "Recent Activity" card
- Tap on the "More Menu"
- Tap on the "Hide This" menu item
- ✅ Verify the view is refreshed and the Recent Activity card is hidden

## Regression Notes
1. Potential unintended areas of impact
The recent activity card is still shown

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing & unit testing

3. What automated tests I added (or what prevented me from doing so)
Add test to `MySiteViewModelTest`

PR submission checklist:

UI Changes testing checklist: N/A
